tests/kernel/semaphore/semaphore/src/main.c

```diff
@@ -20,6 +20,9 @@
 
 #define SEC2MS(s) ((s) * 1000)
 
+/* global variable for mutual exclusion test */
+uint32_t critical_var;
+
```

Comment:
fwiw - Would it be appropriate to put this inside the timeout_info structure? Or is that structure for the thread, and this critical_var is meant for semaphore?

Reply:
Found this helpful tidbit later in the file - "That variable is a critical section and can't be changed by two threads
 * at the same time."

Reply:
I made that variable global, because it imitates the common resource.